client/src/handlers/handlers.ts

```diff
@@ -14,6 +14,189 @@ export function makeHandleRunJob (
   }
 }
 
+let flixFileLocation
+const FLIX_GLOB_PATTERN = '**/*.flix'
+
+/**
+ * It sets the path of flix.jar to a local variable `flixFileLocation` in file `handlers.ts`.
+ * 
+ * @param Filename String (path of the flix.jar)
+ * 
+ * @return void
+*/
+export function setFlixFileName(Filename:String) {
+    flixFileLocation = Filename
+}
+
+/**
+ * returns a count of total active terminals with prefix name `flix`.
+ * 
+ * @return number
+*/
+function countFlixTerminals() {
+    const activeTerminals = vscode.window.terminals
+    let count = 0
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            count+=1
+    }
+    return count
+}
+
+/**
+ * returns an active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
+ *
+ * @return vscode.Terminal
+*/
+function ensureFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
```

Comment:
In modern JS you can write:

```
let colors = ['red', 'green', 'blue'];
for (const color of colors){
    console.log(color);
}
```

see https://stackoverflow.com/questions/3010840/loop-through-an-array-in-javascript

Reply:
same comment applies to countFlixTerminals